pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/elastic/export/painless/UpsertPainless.java

```diff
@@ -41,22 +43,33 @@ public void export() {
         updateByQueryRequest.setMaxRetries(5);
         updateByQueryRequest.setRefresh(true);
         updateByQueryRequest.setAbortOnVersionConflict(false);
+
+        Throwable ioException = null;
         for (int tryNum = 1; tryNum < 3; tryNum++) {
-            if (executeExport(clientInstance, updateByQueryRequest, tryNum)) {
+            ioException = executeExport(clientInstance, updateByQueryRequest, tryNum);
+            if (ioException == null) {
                 break;
             }
         }
+        if (ioException != null) {
+            throw new DsmInternalError("Unable to connect to ElasticSearch", ioException);
```

Comment:
"Problem updating the document database" might be a little better of an error message to throw here so we don't leak internal details of the stack we're using.

Reply:
I think we 'leak' information about ES all over the code, but I get your point. However, the key thing to record here is that we could not connect to ES. I will log an error and remove ES from the message.
I suspect the right way to stop leaking information about internal errors is to catch them at a high level (perhaps in the Spark exception handling), log and strip the content, then send a generic message back to the requester.